modules/geo-layers/src/mvt-layer/mvt-layer.js

```diff
@@ -267,5 +279,27 @@ function transformTileCoordsToWGS84(object, tile, viewport) {
   return feature;
 }
 
+function conversionToWGS84isRequired(tileCoords, isWGS84) {
+  return tileCoords === 'wgs84' && !isWGS84;
+}
+
+function updateTilesWithWGS84Coords(tiles, tilesData, viewport) {
+  tiles.forEach((tile, idx) => {
+    tile.dataWithWGS84Coords = tilesData[idx].map(object =>
+      transformTileCoordsToWGS84(object, tile, viewport)
+    );
+  });
+}
+
+async function convertTilesToWGS84(tiles, viewport) {
+  const tilesData = await Promise.all(
+    tiles.map(async tile => {
+      return (await tile.data) || [];
+    })
+  );
+
+  updateTilesWithWGS84Coords(tiles, tilesData, viewport);
```

Comment:
You can access the loaded data synchronously via `tile.content` once it's loaded. No need to construct a separate list.